odoo/addons/base/views/res_partner_views.xml

```diff
@@ -278,6 +278,7 @@
                                                         <field name="country_id" placeholder="Country" class="o_address_country" options='{"no_open": True, "no_create": True}'/>
                                                     </div>
                                                 </div>
+                                                <field name="vat_child" placeholder="Complete when other legal entity with different VAT" invisible="type == 'contact'"/>
```

Comment:
Instead of introducing this field, can you just use `vat` with a read-only or visible condition?
You probably need to show whether the child is a company as well.

Reply:
@kmagusiak I did the logic that the PO wanted.  If the child contact is of type individual, the VAT field would be the same as the parent, but we show it as empty.  That way if they put a vat, it automatically changes to type company and they are not bothered with whether it is the same vat as the parent, ...  But we can still think further...

Reply:
You have a logic on the form for VAT (sometimes it is read-only), the most obvious is to have the same logic on the form of the child.